components/src/hardware-sim/Labware/labwareInternals/LabwareOutline.tsx

```diff
@@ -28,27 +18,50 @@ export interface LabwareOutlineProps {
   stroke?: CSSProperties['stroke']
   fill?: CSSProperties['fill']
   showRadius?: boolean
+} & (DefinitionProps | DefinitionReplacementProps)
+
+interface DefinitionProps {
+  /** Labware definition to outline */
+  definition: LabwareDefinition
+}
+
+/** Used for rendering an outline without needing a full labware definition. */
+interface DefinitionReplacementProps {
+  definition?: undefined
+  /** x dimension in mm of the outline. */
+  width: number
+  /** y dimension in mm of the outline. */
+  height: number
+  /** minimum x-coordinate (i.e. the left side) of the outline. */
+  minX: number
+  /** minimum y-coordinate (i.e. the bottom) of the outline. */
+  minY: number
 }
 
 const OUTLINE_THICKNESS_MM = 1
 
 export function LabwareOutline(props: LabwareOutlineProps): JSX.Element {
   const {
     definition,
-    width = SLOT_RENDER_WIDTH,
-    height = SLOT_RENDER_HEIGHT,
     isTiprack = false,
     highlight = false,
     highlightShadow = false,
     stroke,
     fill,
     showRadius = true,
   } = props
+
+  const { minX, minY, xDimension, yDimension } =
+    definition !== undefined
```

Comment:
```suggestion
    definition != null
```
Just more of a codebase consistency thing that occasionally saves us some time when refactoring.

Reply:
Ahh, thanks for explaining the advantage.